### PR TITLE
Added endpoint to get facts

### LINKED ISSUE #3134: Add facts api endpoint
Create a v2 endpoint to retrieve all the facts of a resource. In v1 there is a parameters api that allows to also fetch facts. The goal of this ticket is to introduce two API endpoints:

- `/resource/<id>/facts` to list all the facts of a resource
- `/resource/<id>/facts/<fact>` to fetch a single facts from the API

Open questions:
- How do we handle the current rather complex status codes?

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #3134 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Comment by andraskvr** [comment]
I'm not sure about the dto, should it be a `Fact` instead? (then it's sure that the `resource_id` is not an empty string)
Also should the `source` also be specified in the query?

File: changelogs/unreleased/3134-facts-endpoint.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+description: Added endpoint to get facts
+issue-nr: 3132
```
Comment (arnaudsjs) [inline changelogs/unreleased/3134-facts-endpoint.yml:2]: The ticket number in this PR says: `3134`?

File: src/inmanta/protocol/methods_v2.py [hunk 2]
```diff
@@ -670,6 +670,37 @@ def resource_logs(
...
+@typedmethod(
+    path="/resource/<rid>/facts", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2
+)
+def get_facts(tid: uuid.UUID, rid: model.ResourceIdStr) -> List[model.Parameter]:
+    """
+    Get the facts related to a specific resource
+    :param tid: The id of the environment
+    :param rid: Id of the resource
+    :return: The facts related to this resource
+    :raise NotFound: This exception is raised when the referenced environment is not found
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:682]: In the API documentation, the exception name is automatically translated to the corresponding status code. So the description should not talk about an exception.
```suggestion
    :raise NotFound: This status code is returned when the referenced environment is not found
```

File: src/inmanta/protocol/methods_v2.py [hunk 3]
```diff
@@ -670,6 +670,37 @@ def resource_logs(
...
+    api_version=2,
+)
+def get_fact(tid: uuid.UUID, rid: model.ResourceIdStr, id: uuid.UUID) -> model.Parameter:
+    """
+    Get one specific fact
+    :param tid: The id of the environment
+    :param rid: The id of the resource
+    :param id: The id of the fact
+    :return: A specific fact corresponding to the id
+    :raise NotFound: This exception is raised when the referenced environment or fact is not found
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:700]:
```suggestion
    :raise NotFound: This status code is returned when the referenced environment or fact is not found
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/3134-facts-endpoint.yml (hunk 1 above)
Comment (andraskvr) [reply changelogs/unreleased/3134-facts-endpoint.yml]: nice catch

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**